python/lsst/multiprofit/fitting/fit_bootstrap_model.py

```diff
@@ -0,0 +1,269 @@
+    """A catalog fitter that bootstraps a single model.
+
+    This fitter generates a different noisy image of the specified model for
+    each row. The resulting catalog can be used to examine performance and
+    statistics of the best-fit parameters.
+    """
+
+    model_config = frozen_arbitrary_allowed_config
+
+    def get_model_radec(self, source: Mapping[str, Any], cen_x: float, cen_y: float) -> tuple[float, float]:
+        return float(cen_x), float(cen_y)
+
+    def initialize_model(
+        self,
+        model: g2f.ModelD,
+        source: Mapping[str, Any],
+        catexps: list[CatalogExposureSourcesABC],
+        values_init: Mapping[g2f.ParameterD, float] | None = None,
+        centroid_pixel_offset: float = 0,
+    ):
+        if values_init is None:
+            values_init = {}
+        min_x, max_x = np.Inf, -np.Inf
+        min_y, max_y = np.Inf, -np.Inf
+        for idx_obs, observation in enumerate(model.data):
+            x_min = observation.image.coordsys.x_min
+            min_x = min(min_x, x_min)
+            max_x = max(max_x, x_min + observation.image.n_cols * observation.image.coordsys.dx1)
+            y_min = observation.image.coordsys.y_min
+            min_y = min(min_y, y_min)
+            max_y = max(max_y, y_min + observation.image.n_rows * observation.image.coordsys.dy2)
+
+        cen_x = (min_x + max_x) / 2.0
+        cen_y = (min_y + max_y) / 2.0
+
+        params_limits_init = {
+            g2f.CentroidXParameterD: (cen_x, (min_x, max_x)),
+            g2f.CentroidYParameterD: (cen_y, (min_y, max_y)),
+        }
+
+        params_free = get_params_uniq(model, fixed=False)
+        for param in params_free:
+            value_init, limits_new = params_limits_init.get(type(param), (values_init.get(param), None))
+            if value_init is not None:
+                param.value = value_init
+            if limits_new:
+                param.limits.min = -np.Inf
+                param.limits.max = limits_new[1]
+                param.limits.min = limits_new[0]
+
+        # Should be done in get_source_observation, but it gets called first
+        # ... and therefore does not have the initialization above
+        # Also, this must be done per-iteration because PSF parameters vary
+        model.setup_evaluators(evaluatormode=g2f.EvaluatorMode.image)
+        model.evaluate()
+
+        # The offset is to keep the rng seed different from the PSF image seed
+        # It doesn't really need to be so large but it's reasonably safe
+        rng = np.random.default_rng(source["id"] + 10000000)
+
+        for idx_obs, observation in enumerate(model.data):
+            config_obs = catexps[idx_obs].config_boot.observation
+            image, sigma_inv = observation.image, observation.sigma_inv
+            # TODO: This doesn't raise or warn or anything if setting to
+            # the wrong (differently sized output). That seems dangerous.
```

Comment:
Should this be fixed? At the very least please open a ticket.

Reply:
I added asserts. It kind of surprised me that numpy doesn't check this itself.